lib/phoenix_component.ex

```diff
@@ -2014,6 +2014,187 @@ defmodule Phoenix.Component do
   defp form_method(method) when method in ~w(get post), do: {method, nil}
   defp form_method(method) when is_binary(method), do: {"post", method}
 
+  @doc """
+  Renders nested form inputs for associations or embeds.
+
+  [INSERT LVATTRDOCS]
+
+  This function is built on top of `Phoenix.HTML.Form.inputs_for/3`.
+
+  For more information about options and how to build inputs, see `Phoenix.HTML.Form`.
+
+  ## Examples
+
+  ```heex
+  <.form
+    :let={f}
+    for={@changeset}
+    phx-change="change_name"
+  >
+    <.inputs_for :let={f_nested} field={{:f, :nested}}>
+      <%= text_input f_nested, :name %>
+    </.inputs_for>
+  </.form>
+  ```
+  """
+  attr.(:field, :any,
+    required: true,
+    doc: "A %Phoenix.HTML.Form{}/field name tuple, for example: {f, :email}."
+  )
+
+  attr.(:id, :string,
+    doc: """
+    The id to be used in the form, defaults to the concatenation of the given
+    field to the parent form id.
+    """
+  )
+
+  attr.(:as, :atom,
+    doc: """
+    The name to be used in the form, defaults to the concatenation of the given
+    field to the parent form name.
+    """
+  )
+
+  attr.(:default, :any, doc: "The value to use if none is available.")
+
+  attr.(:prepend, :list,
+    doc: """
+    The values to prepend when rendering. This only applies if the field value
+    is a list and no parameters were sent through the form.
+    """
+  )
+
+  attr.(:append, :list,
+    doc: """
+    The values to append when rendering. This only applies if the field value
+    is a list and no parameters were sent through the form.
+    """
+  )
+
+  attr.(:skip_hidden, :boolean,
+    default: false,
+    doc: """
+    Skip the automatic rendering of hidden fields to allow for more tight control
+    over the generated markup. You can access `form.hidden` or use `.hidden_inputs/1`
+    to generate them manually.
+    """
+  )
+
+  slot.(:inner_block, required: true, doc: "The content rendered for each nested form.")
+
+  def inputs_for(assigns) do
+    {form, field} = assigns[:field] || raise ArgumentError, "missing :field assign to inputs_for"
+
+    options =
+      assigns
+      |> Map.take([:id, :as, :default, :append, :prepend])
+      |> Enum.reject(fn {_, v} -> v == nil end)
+
+    options =
+      form.options
+      |> Keyword.take([:multipart])
+      |> Keyword.merge(options)
+
+    assigns =
+      assigns
+      |> assign(:field, nil)
+      |> assign(:forms, form.impl.to_form(form.source, form, field, options))
+
+    ~H"""
+    <%= for finner <- @forms do %>
+      <.hidden_inputs :if={!@skip_hidden} form={finner} />
```

Comment:
Sorry, I meant we should inline the implementation of `.hidden_inputs` directly here. I don't think we need to expose it as a function. :)

Reply:
We can also simplify the implementation of hidden inputs. I don't think we need to generate an ID, that should simplify things. I think the implementation could be something like:

```
<%= for {name, value_or_values} <- finner.hidden,
        name = name_for_value_or_values(value_or_values),
        value <- List.wrap(value_or_values) do %>
  <input type="hidden" name={name} value={value} />
<% end %>
```

and name_for_value_or_values adds `[]` if necessary. :)